### DISCUSSION TITLE
Left and right channel frequency.

### DISCUSSION THREAD

**Discussion by AidansLab** [pschatzmann/arduino-audio-tools#1616] [Q&A] [answered]
I am working on what is basically an MP3 player with a tweeter and subwoofer, and I was wondering if it was possible to use filters on the mp3 files in such a way that one range of frequencies will play on one channel (left or right) and another range on the opposite channel, so that only the bass range will play on the subwoofer, and everything else on the tweeter. I'm just using the player-sd-audiokit example right now.

/**
 * @file player-sd-audiokit.ino
 * @brief see https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-audiokit/player-sd-audiokit/README.md
 * Make sure that the pins are set to off, on, on, off, off
 * @author Phil Schatzmann
 * @copyright GPLv3
 */

#include "AudioTools.h"
#include "AudioLibs/AudioBoardStream.h"
#include "AudioLibs/AudioSourceSD.h" // or AudioSourceIdxSD.h
#include "AudioCodecs/CodecMP3Helix.h"

const char *startFilePath="/";
const char* ext="mp3";
AudioSourceSD source(startFilePath, ext, PIN_AUDIO_KIT_SD_CARD_CS);
AudioBoardStream kit(AudioKitEs8388V1);
MP3DecoderHelix decoder;  // or change to MP3DecoderMAD
AudioPlayer player(source, kit, decoder);

void next(bool, int, void*) {
   player.next();
}

void previous(bool, int, void*) {
   player.previous();
}

void startStop(bool, int, void*) {
   player.setActive(!player.isActive());
}

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  // sd_active is setting up SPI with the right SD pins by calling 
  // SPI.begin(PIN_AUDIO_KIT_SD_CARD_CLK, PIN_AUDIO_KIT_SD_CARD_MISO, PIN_AUDIO_KIT_SD_CARD_MOSI, PIN_AUDIO_KIT_SD_CARD_CS);
  cfg.sd_active = true;
  kit.begin(cfg);

  // setup additional buttons 
  kit.addDefaultActions();
  kit.addAction(kit.getKey(1), startStop);
  kit.addAction(kit.getKey(4), next);
  kit.addAction(kit.getKey(3), previous);

  // setup player
  player.setVolume(0.7);
  player.begin();

}

void loop() {
  player.copy();
  kit.processActions();
}

**Comment by pschatzmann** [comment, marked as answer]
Just specify a FilteredStream that has the kit as output and define different filters on the left and right channels

**Reply by AidansLab** [reply]
How do I go about specifying left and right channels? Is there example code for that?

**Reply by pschatzmann** [reply]
Just read the Wiki about filters
